risc0/zkvm/platform/src/syscall.rs

```diff
@@ -393,3 +404,49 @@ pub unsafe extern "C" fn sys_write(fd: u32, write_buf: *const u8, nbytes: usize)
         nbytes as u32,
     );
 }
+
+/// Retrieves the value of an environment variable, and stores as much
+/// of it as it can it in the memory at [out_words, out_words +
+/// out_nwords).  Returns the length of the value.
+///
+/// This is normally called twice to read an environment variable:
+/// Once to get the length of the value, and once to fill in allocated
+/// memory.
+#[no_mangle]
+pub unsafe extern "C" fn sys_getenv(
+    out_words: *mut u32,
+    out_nwords: usize,
+    varname: *const u8,
+    varname_len: usize,
+) -> usize {
+    let Return(a0, _) = syscall_2(
+        nr::SYS_GETENV,
+        out_words,
+        out_nwords,
+        varname as u32,
+        varname_len as u32,
+    );
+    if a0 == u32::MAX {
+        usize::MAX
+    } else {
+        a0 as usize
+    }
+}
+
+// Number of words remaining in the heap that haven't yet been allocated.
+static mut HEAP_WORDS_REMAINING: usize = crate::memory::HEAP.len_words();
+
+#[no_mangle]
+pub unsafe extern "C" fn sys_alloc_words(nwords: usize) -> *mut u32 {
+    // SAFETY: Single threaded, so nothing else can touch this while we're working.
```

Comment:
Maybe move to a `# Safety` section in the docstr.

Reply:
I think we also need to get into the habit of docstrings for all public functions